planner/core/expression_rewriter.go

```diff
@@ -961,11 +982,18 @@ func (er *expressionRewriter) handleInSubquery(ctx context.Context, v *ast.Patte
 	lt, rt := lexpr.GetType(), rexpr.GetType()
 	collFlag := collate.CompatibleCollate(lt.GetCollate(), rt.GetCollate())
 
+	noDecorrelate := hintFlags&HintFlagNoDecorrelate > 0
+	corCols := extractCorColumnsBySchema4LogicalPlan(np, er.p.Schema())
+	if len(corCols) == 0 && noDecorrelate {
+		er.sctx.GetSessionVars().StmtCtx.AppendWarning(ErrInternal.GenWithStack(
```

Comment:
Should we reset the `noDecorrelate` here?

Reply:
ditto.